### PR TITLE
Making process_watchdog part of manager_param and removed a process_watchdog check to reflect current state of watchdogs in OpenWPM

### LINKED ISSUE #749: Reactivate process_watchdog now that it's fixed
https://github.com/mozilla/OpenWPM/blob/181920fb229b708ec6b4010b9f85dab66d9baff5/automation/TaskManager.py#L69
change this to True?

### PR DESCRIPTION
**Pull request by ankushduacodes**
fixes #749

### CODE REVIEW

File: docs/Platform-Architecture.md [hunk 1]
```diff
@@ -22,11 +22,13 @@ To learn more about the `manager_params` and `browser_params` have a look at [Co
 
 ## Watchdogs
 In OpenWPM we have a so called watchdog that tries to ensure two things.
-- `process_watchdog` 
-    * It is an optional parameter that can be passed to the `TaskManager` to create another thread that kills off `GeckoDriver` (or `Xvfb`) instances that haven't been spawned by OpenWPM. (GeckoDriver is used by Selenium to control Firefox and Xvfb a "virtual display" so we simulate having graphics when running on a server)
+- `process_watchdog`
+    * It is part of default manager_params. It is set to false by default which can manually be set to true.
+    * It is used to create another thread that kills off `GeckoDriver` (or `Xvfb`) instances that haven't been spawned by OpenWPM. (GeckoDriver is used by Selenium to control Firefox and Xvfb a "virtual display" so we simulate having graphics when running on a server).
```
Comment (vringar) [inline docs/Platform-Architecture.md:27]:
```suggestion
    * It is used to create another thread that kills off `GeckoDriver` (or `Xvfb`) instances that aren't currently controlled by OpenWPM. (GeckoDriver is used by Selenium to control Firefox and Xvfb a "virtual display" so we simulate having graphics when running on a server).
```
Usually these instances have been spawned by OpenWPM but we forgot to close them when handling an exception or similiar

**Review by vringar — CHANGES_REQUESTED** [review]
This looks good, except for one small change

File: automation/TaskManager.py [hunk 2]
```diff
@@ -253,8 +245,7 @@ def _manager_watchdog(self) -> None:
 
             # Check for browsers or displays that were not closed correctly
             # 300 second buffer to avoid killing freshly launched browsers
-            # TODO This buffer should correspond to the maximum spawn timeout
-            if self.process_watchdog:
+            if self.manager_params[PROCESS_WATCHDOG]:
```
Comment (vringar) [inline automation/TaskManager.py:248]:
```suggestion
             # TODO This buffer should correspond to the maximum spawn timeout
            if self.manager_params[PROCESS_WATCHDOG]:
```

This TODO still exists. The one that can be removed is further up.

Comment (ankushduacodes) [reply automation/TaskManager.py:248]: I have removed it in latest push 12 mins ago. Is this possible that you may have check outdated commit? Please let me know if it is still there in latest commit

Comment (vringar) [reply automation/TaskManager.py:248]: I just refreshed and GitHub claims it's still present in the latest version of this PR.

Comment (ankushduacodes) [reply automation/TaskManager.py:248]: Okay, I will commit this suggestion.

**Comment by ankushduacodes** [comment]
@vringar I think I forgot to run pre-commit before committing and pushing this branch and because of which it failed to run the tests. Is there a way to run the tests again or do I have to make change to make tests run again?

**Comment by vringar** [comment]
@ankushduacodes I'd suggest running `pre-commit run --all` and then commit the changes made and push those. The new commit should make the tests rerun.

**Comment by ankushduacodes** [comment]
> @ankushduacodes I'd suggest running `pre-commit run --all` and then commit the changes made and push those. The new commit should make the tests rerun.

@vringar There were no changes made after I ran `pre-commit run --all` on my local machine.

**Comment by ankushduacodes** [comment]
Hmmm, I am not sure what is happening, even if I remove the TODO comment, it seem to somehow come back... This is third time I have removed it.... 🤔🤔

**Comment by vringar** [comment]
I'm looking at the commit history and can't explain it to myself either. If figuring this out is too much of a headache I can also merge it into a temporary branch, make the changes needed and then merge it myself.

**Comment by ankushduacodes** [comment]
> I'm looking at the commit history and can't explain it to myself either. If figuring this out is too much of a headache I can also merge it into a temporary branch, make the changes needed and then merge it myself.

For sure, If you need me to do something I will be happy to do it. 😄

**Comment by ankushduacodes** [comment]
I think latest commit did detele, but I should mention that I am not sure what was causing this issue as my both this branch and my forked master branch are even with mozilla/OpenWPM master branch

File: automation/TaskManager.py (hunk 2 above)
Comment (vringar) [reply automation/TaskManager.py:245]: Can you commit this change again? Seems like it got lost along the way.

**Review by vringar — CHANGES_REQUESTED** [review]

File: __init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+# __init__.py
```
Comment (vringar) [inline __init__.py:1]: Also please please remove this comment
```suggestion
```

Comment (ankushduacodes) [reply __init__.py:1]: Ah... yes, I had added this comment just to make the tests the tests run. I intended to remove it but forgot, Thank you for reminding me

**Comment by ankushduacodes** [comment]
I think the comment came back!...😲

**Comment by ankushduacodes** [comment]
I think I got it, I was removing the wrong comment by accident and When you were suggesting a change I was coming back with that change. I totally mixed two comments without looking into it too much, I am sorry for all the confusion 😅

**Comment by vringar** [comment]
No worries. I'm just going to wait for the tests to pass and then merge this

File: automation/TaskManager.py [hunk 4]
```diff
@@ -253,8 +242,7 @@ def _manager_watchdog(self) -> None:
...
```
Comment (vringar) [inline automation/TaskManager.py:247]:
```suggestion
            # TODO This buffer should correspond to the maximum spawn timeout
            if self.manager_params[PROCESS_WATCHDOG]:
```

File: automation/TaskManager.py [hunk 5]
```diff
@@ -254,7 +243,9 @@ def _manager_watchdog(self) -> None:
             # Check for browsers or displays that were not closed correctly
             # 300 second buffer to avoid killing freshly launched browsers
             # TODO This buffer should correspond to the maximum spawn timeout
-            if self.process_watchdog:
+             # TODO This buffer should correspond to the maximum spawn timeout
```
Comment (vringar) [inline automation/TaskManager.py:246]:
```suggestion
```

**Review by vringar — APPROVED** [review]

**Pull request merged**